File: type-script-nest-graph-ql-실습/src/app/common/constant/page.request.dto.ts
import { Field, InputType } from "@nestjs/graphql";

@InputType()
export abstract class PageRequestDto {
  @Field(() => Number)
  pageNumber?: number;

  @Field(() => Number)
  perPageSize?: number;

  @Field(() => Boolean)
  orderBy?: boolean;

  getOffset(): number {
    return (this.pageNumber - 1) * this.perPageSize;
  }

  getLimit(): number {
    return this.perPageSize;
  }
}

File: kotlin-graphql-test/src/main/kotlin/com/junyharangstudy/kotlingraphqltest/common/constant/PageRequestDto.kt
package com.junyharangstudy.kotlingraphqltest.common.constant

class PageRequestDto(
    var pageNumber: Long? = null,
    var perPageSize: Long? = null,
    var orderBy: Boolean? = null
) {

    fun getOffset(): Long {
        return((pageNumber?.minus(1) ?: 0) * (perPageSize ?: 0))
    }

    fun getLimit(): Long {
        return perPageSize ?: 0
    }
}